src/pykeen/utils.py

```diff
@@ -146,9 +146,12 @@ def resolve_device(device: DeviceHint = None) -> torch.device:
         device = "cuda"
     if isinstance(device, str):
         device = torch.device(device)
-    if not torch.cuda.is_available() and device.type == "cuda":
+    if not torch.cuda.is_available() and not torch.backends.mps.is_available() and device.type == "cuda":
```

Comment:
`torch.backends.mps` appears only in torch 1.12 and later, anything older throws an error `AttributeError: module 'torch.backends' has no attribute 'mps'`. 
So this line right now is not backward-compatible, but we might find a workaround

Reply:
just define a helper function that wraps this in a try/except AttributeError that returns false on an error

Reply:
This isn't a problem anymore since we require torch 2.0 minimum now